packages/xstate-react/src/useActor.ts

```diff
@@ -1,16 +1,37 @@
-import { useCallback } from 'react';
-import { ActorRef, EventObject, SnapshotFrom } from 'xstate';
-import { useSyncExternalStore } from 'use-sync-external-store/shim';
-
-export function useActor<TActor extends ActorRef<any, any>>(
-  actorRef: TActor
-): [SnapshotFrom<TActor>, TActor['send']];
-export function useActor<TEvent extends EventObject, TSnapshot>(
-  actorRef: ActorRef<TEvent, TSnapshot>
-): [TSnapshot, (event: TEvent) => void];
-export function useActor(
-  actorRef: ActorRef<EventObject, unknown>
-): [unknown, (event: EventObject) => void] {
+import { useCallback, useEffect, useState } from 'react';
+import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/shim/with-selector';
+import {
+  ActorRefFrom,
+  AnyActorBehavior,
+  AnyState,
+  InterpreterOptions,
+  InterpreterStatus,
+  SnapshotFrom
+} from 'xstate';
+import { useIdleInterpreter } from './useActorRef.ts';
+
+function identity<T>(a: T): T {
+  return a;
+}
+
+const isEqual = (prevState: AnyState, nextState: AnyState) => {
+  return prevState === nextState || nextState.changed === false;
+};
```

Comment:
Note that this one doesn't belong here - yet another reason why we need to get rid of `.changed` 😉

Reply:
(that will be done in a separate batch of work though, let's leave it as is here)